splunk-otel-android-volley/src/main/java/com/splunk/rum/VolleyHttpClientAttributesGetter.java

```diff
@@ -68,18 +70,27 @@ public String method(RequestWrapper requestWrapper) {
 
     @Override
     public List<String> requestHeader(RequestWrapper requestWrapper, String name) {
-        String header;
+        Request<?> request = requestWrapper.getRequest();
         try {
-            header = requestWrapper.getRequest().getHeaders().get(name);
-            if (header == null) {
-                header = requestWrapper.getAdditionalHeaders().get(name);
-            }
-
+            Map<String, String> headers = request.getHeaders();
+            Map<String, String> additionalHeaders = requestWrapper.getAdditionalHeaders();
+            List<String> result = new ArrayList<>(headers.size() + additionalHeaders.size());
+            result.addAll(findCaseInsensitive(name, headers));
```

Comment:
Nit: I think in the usual scenario there'll be much less header values than the total number of headers in the request:
```suggestion
            List<String> result = new ArrayList<>();
            result.addAll(findCaseInsensitive(name, headers));
```
Also, you could pass a reference to the `result` list to the `findCaseInsensitive` method to avoid allocating 2 more lists.

Reply:
> I've always considered mutable parameters to be an antipattern/bad practice. I'd also call this premature optimization...

They're absolutely terrible in a public API, I 100% agree.
I sometimes use them in private, hidden away code, in some cases it's a useful trick.

Reply:
I went ahead and merged and will happily review a follow-up PR if you choose to submit one. 🙃